src/libasr/pass/intrinsic_functions.h

```diff
@@ -1464,7 +1464,11 @@ namespace Dshiftl {
             append_error(diag, "The shift argument of 'dshiftl' intrinsic must be non-negative integer", loc);
             return nullptr;
         }
-        int k_val = (kind == 8) ? 64: 32;
+        int k_val = kind * 8;
+        if (shift > k_val) {
+            append_error(diag, "The shift argument of 'dshiftl' intrinsic must be less than or equal to the bit size of the integer", loc);
```

Comment:
```suggestion
            append_error(diag, "The shift argument of 'dshiftl' intrinsic must be less than or equal to the bit size of the integer", args[2]->base.loc);
```
or something like that, this will make the squiggle point to just args[2]

Reply:
+1
@Jaysukh-409 thanks for the PR. Can you please do the same change for dshiftr as well? That would be helpful.

Reply:
For dshiftr, We don't need to change anything. Because we already handled error messages there.